remc2/engine/Basic.cpp

```diff
@@ -2026,9 +2042,13 @@ void DrawLineHighRes(int16_t posStartX, int16_t posStartY, int16_t posEndX, int1
 	int v9; // ecx
 	int v10; // [esp+0h] [ebp-4h]
 
-	v5 = (x_BYTE*)(pitch * posStartY + pdwScreenBuffer + posStartX);
+	int helpWidth = 640;
+	if (!DefaultResolutions())
+		helpWidth = screenWidth_18062C;
+
+	v5 = (x_BYTE*)(helpWidth * posStartY + pdwScreenBuffer_351628 + posStartX);
 	v6 = posEndY;
-	v10 = (unsigned __int16)(pitch - posEndX);
+	v10 = (unsigned __int16)(helpWidth - posEndX);
 	if (x_WORD_E36D4 & 4)
```

Comment:
I am not sure about the name "helpWidth". Pitch is graphics term for basically line resolution in the buffer. We could use method overloading here? We can check DefaultResolutions in the overload method before calling the original method with a fix pitch of 640?